narwhals/_pandas_like/series.py

```diff
@@ -138,6 +141,8 @@ def _from_iterable(
         )
 
     def __len__(self) -> int:
+        if self._implementation == "dask":
+            return len(self._series)
         return self.shape[0]
```

Comment:
we might as well just return `len(self._series)` for all of them, thus avoiding the if statement?